receiver/awscontainerinsightreceiver/internal/k8swindows/extractors/fs_extractor.go

```diff
@@ -0,0 +1,72 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package extractors // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/cadvisor/extractors"
+
+import (
+	ci "github.com/open-telemetry/opentelemetry-collector-contrib/internal/aws/containerinsight"
+	awsmetrics "github.com/open-telemetry/opentelemetry-collector-contrib/internal/aws/metrics"
+	cExtractor "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/cadvisor/extractors"
+
+	"go.uber.org/zap"
+)
+
+type FileSystemMetricExtractor struct {
+	logger         *zap.Logger
+	rateCalculator awsmetrics.MetricCalculator
+}
+
+func (f *FileSystemMetricExtractor) HasValue(rawMetric RawMetric) bool {
+	if !rawMetric.Time.IsZero() {
+		return true
+	}
+	return false
+}
+
+func (f *FileSystemMetricExtractor) GetValue(rawMetric RawMetric, mInfo cExtractor.CPUMemInfoProvider, containerType string) []*cExtractor.CAdvisorMetric {
+	if containerType == ci.TypePod {
+		return nil
+	}
+
+	containerType = getFSMetricType(containerType, f.logger)
+	metrics := make([]*cExtractor.CAdvisorMetric, 0, len(rawMetric.FileSystemStats))
+
+	for _, v := range rawMetric.FileSystemStats {
+		metric := cExtractor.NewCadvisorMetric(containerType, f.logger)
+
+		metric.AddField(ci.MetricName(containerType, ci.FSUsage), v.UsedBytes)
+		metric.AddField(ci.MetricName(containerType, ci.FSCapacity), v.CapacityBytes)
+		metric.AddField(ci.MetricName(containerType, ci.FSAvailable), v.AvailableBytes)
+
+		if v.CapacityBytes != 0 {
+			metric.AddField(ci.MetricName(containerType, ci.FSUtilization), float64(v.UsedBytes)/float64(v.CapacityBytes)*100)
+		}
+
+		metrics = append(metrics, metric)
+	}
+	return metrics
+}
+
+func (f *FileSystemMetricExtractor) Shutdown() error {
+	return f.rateCalculator.Shutdown()
+}
+
+func NewFileSystemMetricExtractor(logger *zap.Logger) *FileSystemMetricExtractor {
+	return &FileSystemMetricExtractor{
+		logger:         logger,
+		rateCalculator: cExtractor.NewFloat64RateCalculator(),
+	}
+}
+
+func getFSMetricType(containerType string, logger *zap.Logger) string {
```

Comment:
We can refactor this to 
```
	switch containerType {
	case ci.TypeNode:
		return  ci.TypeNodeFS
	case ci.TypeContainer:
		return ci.TypeContainerFS
	default:
		logger.Warn("fs_extractor: fs metric extractor is parsing unexpected containerType", zap.String("containerType", containerType))
return ""
```

Reply:
You suggested good alternative implementation but i am trying to keep this implementation consistent to cadavisor fs extractor.